static/app/views/onboarding/setupDocsLoader.tsx

```diff
@@ -0,0 +1,240 @@
+import {Fragment, useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+import {motion} from 'framer-motion';
+import {Location} from 'history';
+import beautify from 'js-beautify';
+
+import {Button} from 'sentry/components/button';
+import {CodeSnippet} from 'sentry/components/codeSnippet';
+import ExternalLink from 'sentry/components/links/externalLink';
+import LoadingError from 'sentry/components/loadingError';
+import {DocumentationWrapper} from 'sentry/components/onboarding/documentationWrapper';
+import {PRODUCT, ProductSelection} from 'sentry/components/onboarding/productSelection';
+import {PlatformKey} from 'sentry/data/platformCategories';
+import platforms from 'sentry/data/platforms';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Organization, Project, ProjectKey} from 'sentry/types';
+import handleXhrErrorResponse from 'sentry/utils/handleXhrErrorResponse';
+import useApi from 'sentry/utils/useApi';
+import SetupIntroduction from 'sentry/views/onboarding/components/setupIntroduction';
+import {DynamicSDKLoaderOption} from 'sentry/views/settings/project/projectKeys/details/keySettings';
+
+export function SetupDocsLoader({
+  organization,
+  location,
+  project,
+  platform,
+  close,
+}: {
+  close: () => void;
+  location: Location;
+  organization: Organization;
+  platform: PlatformKey | null;
+  project: Project;
+}) {
+  const api = useApi();
+  const currentPlatform = platform ?? project?.platform ?? 'other';
+  const [projectKey, setProjectKey] = useState<ProjectKey | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [projectKeyUpdateError, setProjectKeyUpdateError] = useState(false);
+
+  const fetchData = useCallback(async () => {
+    const keysApiUrl = `/projects/${organization.slug}/${project.slug}/keys/`;
+
+    try {
+      const loadedKeys = await api.requestPromise(keysApiUrl);
+
+      if (loadedKeys.length === 0) {
+        setHasLoadingError(true);
+        return;
+      }
+
+      setProjectKey(loadedKeys[0]);
+      setHasLoadingError(false);
+    } catch (error) {
+      setHasLoadingError(error);
+      throw error;
+    }
+  }, [api, organization.slug, project.slug]);
+
+  // Automatically update the products on the project key when the user changes the product selection
+  // Note that on initial visit, this will also update the project key with the default products (=all products)
+  // This DOES NOT take into account any initial products that may already be set on the project key - they will always be overwritten!
+  const updateSelectedProducts = useCallback(async () => {
+    const productsQuery =
+      (location.query.product as PRODUCT | PRODUCT[] | undefined) ?? [];
+    const products = typeof productsQuery === 'string' ? [productsQuery] : productsQuery;
```

Comment:
we can use the function https://github.com/getsentry/sentry/blob/7dd5aa6ba15a703fad25766356dccf4018ed6a03/static/app/utils/queryString.tsx#L98 here 😉